scripts/wasmtestreport.py

```diff
@@ -695,6 +817,109 @@ def run_subprocess(cmd, label="", cwd=None, shell=False, timeout=None):
+            test_case for test_case in test_cases
+            if should_run_file(test_case, config['run_folders_paths'], 
+                               config['skip_folders_paths'], config['skip_test_cases'])
+        ]
+    
+    return config
+
+# ----------------------------------------------------------------------
+# Function: setup_test_file_in_artifacts
+#
+# Purpose:
+#   Setup a test file in the artifacts directory, centralizing file setup logic
+#
+# Variables:
+# - Input: original_source - path to original test file, artifacts_root - artifacts directory
+# - Output: dest_source - path to test file in artifacts directory
+# ----------------------------------------------------------------------
+def setup_test_file_in_artifacts(original_source, artifacts_root):
+    """Setup a test file in the artifacts directory"""
+    try:
+        rel_path = original_source.relative_to(TEST_FILE_BASE)
+    except ValueError:
+        rel_path = Path(original_source.name)
+    
+    dest_dir = artifacts_root / rel_path.parent
+    dest_dir.mkdir(parents=True, exist_ok=True)
+    dest_source = dest_dir / original_source.name
+    
+    # Create symlink or copy file
+    if not dest_source.exists():
+        try:
+            dest_source.symlink_to(original_source)
+        except OSError:
+            shutil.copy2(original_source, dest_source)
+    
+    # Copy expected outputs directory if present
+    expected_dir_src = original_source.parent / EXPECTED_DIRECTORY
+    if expected_dir_src.is_dir():
+        expected_dir_dst = dest_dir / EXPECTED_DIRECTORY
+        if not expected_dir_dst.exists():
+            shutil.copytree(expected_dir_src, expected_dir_dst)
+    
+    return dest_source
+
+# ----------------------------------------------------------------------
+# Function: run_tests
+#
+# Purpose:
+#   Execute all tests, centralizing test execution logic
+#
+# Variables:
+# - Input: config - test configuration, artifacts_root - artifacts directory, 
+#          results - results dictionary, timeout_sec - timeout for tests
+# - Output: None (modifies results dictionary)
+# ----------------------------------------------------------------------
+def run_tests(config, artifacts_root, results, timeout_sec):
+    """Execute all tests"""
+    total_count = len(config['tests_to_run'])
+    
+    for i, original_source in enumerate(config['tests_to_run']):
+        logger.info(f"[{i+1}/{total_count}] {original_source}")
+        
+        dest_source = setup_test_file_in_artifacts(original_source, artifacts_root)
+        
+        # Determine test type and run appropriate test
+        parent_name = original_source.parent.name
+        if parent_name == DETERMINISTIC_PARENT_NAME:
+            test_single_file_deterministic(dest_source, results["deterministic"], timeout_sec)
+        elif parent_name == NON_DETERMINISTIC_PARENT_NAME:
+            test_single_file_non_deterministic(dest_source, results["non_deterministic"], timeout_sec)
+
```

Comment:
If a test file lives in a folder other than deterministic or non_deterministic, it silently does nothing. This can make mis-tagged or malformed tests vanish without any feedback.

I’d recommend adding an else branch that logs a warning but continues execution.

Maybe just log it as an unknown file and keep going?